src/controller/CHIPDeviceController.cpp

```diff
@@ -961,17 +962,23 @@ CHIP_ERROR DeviceCommissioner::OnOperationalCertificateSigningRequest(NodeId nod
 {
     ReturnErrorCodeIf(mOperationalCredentialsDelegate == nullptr, CHIP_ERROR_INCORRECT_STATE);
 
-    auto opCertBuf     = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(kMaxCHIPOpCertLength));
-    auto opCert        = std::unique_ptr<uint8_t>(opCertBuf);
+    chip::Platform::ScopedMemoryBuffer<uint8_t> opCert;
+
+    opCert.Alloc(kMaxCHIPOpCertLength);
+    ReturnErrorCodeIf(!opCert, CHIP_ERROR_NO_MEMORY);
```

Comment:
nit: can also do

```cpp
ReturnErrorCodeIf(!opCert.Alloc(kMaxCHIPOpCertLength), CHIP_ERROR_NO_MEMORY);
```
 
 to be more concice. Not sure if that improves or lessens readability though :/

Reply:
Updated the code for this and signingCert buffer check.